### PR TITLE
Feature/manater dashboard

### PR DESCRIPTION
**Pull request by saraho1123**
## Is this a fix or a feature?
feature (just realized I spelled the branch with manater, not manager! oops!)

## What is the change?
Added functionality for manager dashboard to be displayed with today's revenue, today's percent rooms booked and today's available rooms.

## What does it fix?
** Had to move two methods from Guest to Hotel so that Manager could access them!

## Where should the reviewer start?
line 23 of Hotel.js
line 291 of base.scss
line 72 of index.html
line 19 of index.js

## How should this be tested?
with repo cloned:
run: npm start in terminal
open: http://localhost:8080/ in browser
login using either manager or customer(# 01-50) and password of 'Overlook2020'.
Using any numbers <01 or >50 should cause an error message to appear below the submit button.
Once correctly logged in, the dashboard for the manager should appear.

## Relevant Tickets:
#16

### CODE REVIEW

File: src/Guest.js [hunk 1]
```diff
@@ -45,23 +45,23 @@ class Guest extends Hotel {
...
 
-  listOccupiedRoomsByDate(date) {
-    const bookedRoomsOnDate = this.bookings.filter(booking => {
-      return booking.date === date;
-    })
-    return bookedRoomsOnDate.map(booked => {
-      return booked.roomNumber
-    })
-  }
+  // listOccupiedRoomsByDate(date) {
```
Comment (saraho1123) [inline src/Guest.js:48]: Need to delete these

File: src/Hotel.js [hunk 2]
```diff
@@ -20,12 +20,32 @@ class Hotel {
...
+    const availableRoomsOnDate = this.rooms.reduce((available, room) => {
+      !bookedRoomNumbers.includes(room.number) ? available.push(room) : null;
+      return available
+    }, [])
+    return availableRoomsOnDate;
+  }
+
   bookRoomForGuest(userID, date, roomNumber) { 
     let bookingData =  { userID: userID, date: date, roomNumber: roomNumber }
     // call fetchPOST here, I think
```
Comment (saraho1123) [inline src/Hotel.js:43]: excited to start POST and DELETE!

File: src/index.html [hunk 3]
```diff
@@ -208,33 +183,35 @@ <h1 class="room-is-booked heading">
...
+        <h2 class="daily-rooms-available heading2">Rooms available for <span>yyyy/mm/dd</span>:</h1>
         <section class="list-rooms-available cards-of-rooms">
-              <h2 class="heading2">Rooms available for <span>yyyy/mm/dd</span>:</h1>
-              <article class="room booked-room">
-                <article class="rooms-available-cards room-details">
+              <!-- <!-- <article class="room vacant-room">
+                <section class="rooms-available-cards room-details">
                   <h2 class="room-number-type">Room 1: Residential Suite</h2>
-                  <section class="small-room-details">
+                  <article class="small-room-details">
```
Comment (saraho1123) [inline src/index.html:205]: remember to delete the HTML you no longer need!

File: src/index.js [hunk 4]
```diff
@@ -181,6 +191,58 @@ function displayGuestPastBookingsDasboard() {
   })
 }
 
+function displayManagerDasboard() {
```
Comment (saraho1123) [inline src/index.js:194]: I think that all my display methods can go in the dom-updates file!

**Review by saraho1123 — COMMENTED** [review]
Making progress!

File: src/Hotel.js [hunk 5]
```diff
@@ -20,12 +20,32 @@ class Hotel {
...
+    })
+    return bookedRoomsOnDate.map(booked => {
+      return booked.roomNumber
+    })
+  }
+
+  listVacantRoomsByDate(date) {
+    const bookedRoomNumbers = this.listBookedRoomNumberByDate(date)
+    const availableRoomsOnDate = this.rooms.reduce((available, room) => {
+      !bookedRoomNumbers.includes(room.number) ? available.push(room) : null;
```
Comment (atownse) [inline src/Hotel.js:35]: nice ternary but if you did an if statement, you wouldn't need another condition for null

File: src/index.js [hunk 6]
```diff
@@ -105,9 +107,8 @@ function validateUserLogin(event) {
     guest = new Guest(usersData, roomsData, bookingsData);
     getGuest();
     enableGuestHomeView();
-    displayGuestNameDasboard();
-    displayGuestPastBookingsDasboard();
-  } else if (userLogin.value === 'manager' && userPassword.value === 'Overlook2020') {
+  } else if (userLogin.value === 'm' && userPassword.value === 'o') {
```
Comment (atownse) [inline src/index.js:110]: Is this for ease while working?

File: src/index.js [hunk 7]
```diff
@@ -181,6 +191,58 @@ function displayGuestPastBookingsDasboard() {
...
+
+function displayVacantRoomsByDate() {
+  //need to get today's bookings!!
+  let date = getToday();
+  let vacantRooms = manager.listVacantRoomsByDate(date);
+  console.log(vacantRooms)
+  vacantRooms.forEach(room => {
+    console.log(room)
+    currentRoomsAvailable.insertAdjacentHTML('afterbegin', `
+      <article class="room vacant-room">
```
Comment (atownse) [inline src/index.js:229]: Will this go in your dom update file?

**Review by atownse — APPROVED** [review]

**Pull request merged**